File: September Challenge/Week 2/Java/Day_10/BullsAndCows.java
// Problem : https://leetcode.com/problems/bulls-and-cows/
// Solution:

class Solution {
    public String getHint(String secret, String guess) {
        int bull=0;
        int cows=0;
        char[] s=secret.toCharArray();
        char[] g=guess.toCharArray();
        int[] arr = new int[10]; 
        for(int i=0;i<s.length;i++){
            if(s[i]==g[i]){
                bull++;
            }else{
                if(arr[s[i]-'0']++ < 0) cows++;
                if(arr[g[i]-'0']-- > 0) cows++;
            }
        }
        return bull+"A"+cows+"B";
    }
}

File: September Challenge/Week 2/Cpp/DAY_10/BullsAndCows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
        vector<int> v(10,0);
        int bulls=0,cows=0;
        for(int i=0;i<secret.length();i++){
            int s = (int)(secret[i]-'0');
            if(secret[i]==guess[i]){
                bulls++;
            }
            v[s]++;
        }
        for(int i=0;i<guess.length();i++){
            int g = (int)(guess[i]-'0');
            if(v[g]>0){
                v[g]--;
                cows++;
            }
        }
        cows=cows-bulls;
        string s =to_string(bulls) + "A" + to_string(cows) + "B";
        return s;
        
    }
};
